Draft blog post on FlowFuse Certified Nodes (#5152)
FlowFuse/website: src/blog/2026/06/why-flowfuse-certified-nodes.md

---
title: "Why FlowFuse Certified Nodes?"

> subtitle: "When vetted, secure, and supported nodes are the right call — and when they aren't"

Reviewer: ```suggestion
subtitle: "Depend on maintained, secure, and supported nodes to reliably integrate your systems."
```

One key point is that we also (co-)develop these nodes. That's not highlighted now. Not sure

> description: "FlowFuse Certified Nodes take the vetting, security monitoring, and support burden of community nodes off your plate. Here are four situations where they're the clear solve."

Reviewer: ```suggestion
description: "FlowFuse Certified Nodes take the vetting, security monitoring, and support burden of community nodes off your plate. Read which four situations where they're the clear solve."
```

  - question: "Which FlowFuse tiers include Certified Nodes?"
    answer: "FlowFuse Certified Nodes are available to Teams and Enterprise tier customers. New instances get automatic access to the catalogue, and you can contact FlowFuse to add Certified Nodes to an existing instance."
tldr: "Every node in a flow is code running in your environment, and community nodes carry hidden security, compatibility, and maintenance risk you have to manage yourself. FlowFuse Certified Nodes take that work on through a defined quality, security, and support process. They're the clear solve for four situations — high-security deployments, regulated environments, uptime-critical operations, and teams that can't afford to debug alone — and they're available to Teams and Enterprise tier customers."
cta:
  type: contact
  title: "Not sure if Certified Nodes fit your deployment?"
  description: "See how FlowFuse Certified Nodes take the vetting, security monitoring, and support burden off your team. Talk to us about your setup and which use case describes you."
---

# Why FlowFuse Certified Nodes?

> With every flow that you build on FlowFuse, you’re using nodes to execute code in your environment. These nodes often connect to your data, your network, and your operations - and while the flexibility and freedom these nodes offer are part of the power behind FlowFuse, the reality of adoption means that you’re inheriting functionality as much as you are inheriting risk.

Reviewer: Maybe we explain these are open source, built by people that assume no liability?

Author: "With every flow that you build on FlowFuse, you’re using nodes to execute code in your environment. These nodes often connect to your data, your network, and your operations - and while the flexibility and freedom these nodes offer are part of the power behind FlowFuse, the reality of adoption means that you’re inheriting functionality as much as you are inheriting risk. Open source development is often offered by people who assume no liability. They needed a solution, they developed the solution, and it worked, but when it comes to adoption, the phrase "caveat emptor" - buyer beware - applies."

For most users, this risk is minimal - but for some orgs, the idea of inheriting risk is a step too far. When an organisation cannot have a day of downtime following a code update, or a potential risk vector unaddressed after a maintainer loses interest, something more is necessary - and FlowFuse Certified Nodes presents a solution to this problem.

FlowFuse Certified nodes exist to take that liability off your hands, delivering functionality and peace of mind. A Certified Node is one that FlowFuse has put through an intense quality and security process, ensuring that it meets form and function before you ever install it. And once it is installed, FlowFuse offers long-term support and critical review capability - meaning that the vetting, monitoring, and maintenance you’d otherwise need to carry becomes something that FlowFuse can solve for you.

## What "Certified" Actually Means

FlowFuse certifies a node against three pillars, each aimed at a specific failure mode.

> **Quality:** Every Certified Node is tested for operational reliability and compatibility, so it behaves predictably across versions instead of surprising you in production. This means no more guesswork, no more deep vetting, and significantly quicker time to value.

Reviewer: We will pick best in class OSS nodes, sub-license best in class proprietary nodes, and build our own integrations. Whatever is needed to have the best in class integration, all in one catalog. Without new vendors for them to select

> **Security:** FlowFuse proactively resolves potential vulnerabilities and revokes certification from any node that falls short - and notifies affected customers when that happens, so you're never the last to find out. Your nodes are secure by default - removing an entire headache from your workflow.

Reviewer: How can one proactively fix security vulnerabilities? I would steer more towards a SLA bound fix, back ported to older versions deployed too. Helping customers reduce stress when fixes are time sensitive, and make it a smooth 5-min task to complete.

> **Support:** When a Certified Node gives you trouble, you get real troubleshooting help and a defined path to resolution, not an unanswered issue queue. FlowFuse node certification also means FlowFuse supported - so you get real, human support, not automatically generated documentation that may or may not map to solutions.

Reviewer: I think the long term goal is to support each minor semver version for a decade. We cannot promise that today, but if you've built something now, it should continue to work and improve long term. (Performance etc)

Author: Is that something we want to call out here, or do we want to be somewhat general in our public stance to give us flexibility?

Let’s take a look at some specific use cases where Certified Nodes are game changers.

## Use Case 1: High-Security and High-Privacy Deployments

> For teams running flows that touch personal health records, customer payment data, or proprietary process telemetry, every node in that flow executes with access to sensitive data. This means that every dependency is a potential way in - and a potential source of regulatory fines and intervention.

Reviewer: Military production! 

I wouldn't discuss personal health records, FF hardly processes these. And opens a can of worms about compliance. Pharma manufacturing, yes!

---

> title: "Why FlowFuse Certified Nodes?"

Reviewer: ```suggestion
title: "The what and the why: FlowFuse Certified Nodes"
```

subtitle: "Depend on maintained, secure, and supported nodes to reliably integrate your systems."
description: "FlowFuse Certified Nodes take the vetting, security monitoring, and support burden of community nodes off your plate. Read which four situations where they're the clear solve."

> date: 2026-06-05

Reviewer: ```suggestion
date: 2026-06-17
```

For most users, this risk is minimal - but for some orgs, the idea of inheriting risk is a step too far. When an organisation cannot have a day of downtime following a code update, or a potential risk vector unaddressed after a maintainer loses interest, something more is necessary - and FlowFuse Certified Nodes presents a solution to this problem.

FlowFuse Certified nodes exist to take that liability off your hands, delivering functionality and peace of mind. A Certified Node is one that FlowFuse has put through an intense quality and security process, ensuring that it meets form and function before you ever install it. And once it is installed, FlowFuse offers long-term support and critical review capability - meaning that the vetting, monitoring, and maintenance you’d otherwise need to carry becomes something that FlowFuse can solve for you.

## What "Certified" Actually Means

FlowFuse certifies a node against three pillars, each aimed at a specific failure mode.

> **Quality:** Every Certified Node is tested for operational reliability and compatibility, so it behaves predictably across versions instead of surprising you in production. This means no more guesswork, no more deep vetting, and significantly quicker time to value. We pick only the best-in-class OSS nodes, sub-license best-in-class properietary nodes, and build custom integrations for FlowFuse users. This means you get true best-in-class integrations in a single catalog - with no need to select new vendors or navigate complex licenses.

Reviewer: This is a bit too much AI for my taste. The bold combined with the `:`, it's offputting to me

Author: This was hand-generated so worrisome if it comes across that way. How does it read now - it has subheaders in h3.

## How to Know If This Is You

These four situations aren't mutually exclusive - and they’re by no means exhaustive. A hospital system can carry all four at once - security, compliance, uptime, and support - as well as a complex maze of additional operational requirements and restrictions. FlowFuse Certified Nodes are meant to resolve this complexity and deliver a powerful implementation at scale, resulting in a stack that is deployable, maintainable, and - critically - trustworthy.

If your flows run on a bench or power a low-stakes internal tool, the open library serves you well, and managing your own vetting is a perfectly reasonable choice. Not every team needs Certified Nodes, and we won't pretend otherwise. 

But if your flows touch sensitive data, fall under industry regulations, carry any uptime obligations, or back a team that can't afford to debug alone, then the vetting, monitoring, and support you'd otherwise have to build yourself is exactly what FlowFuse provides. The value is not just in the nodes themselves - it's in handing off work that was quietly becoming yours by default.

> ## Certified Ndoes Today

Reviewer 2: Are we going to talk about previously certified nodes as well? Also, since this blog will be published alongside the launch of the OPC UA node, I think we should cover that and remove the reference to `node-red-contrib-opcua`, as that's not the node we're certifying. The certification is for Sterfive's OPC UA node, so we should make that clear in the blog

Reviewer 2: for more information: https://flowfuse.slack.com/archives/C0A91GJ4221/p1781602310522699

Author: Yes - this was waiting for final name/documentation. Updated now that we have that.

---
title: "The what and the why: FlowFuse Certified Nodes"
subtitle: "Depend on maintained, secure, and supported nodes to reliably integrate your systems."
description: "FlowFuse Certified Nodes take the vetting, security monitoring, and support burden of community nodes off your plate. Read which four situations where they're the clear solve."

> date: 2026-06-26

Reviewer 2: Date needs to be updated

> date: 2026-06-26

Reviewer 2: you need to move blog to current month folder : `2026/07/`

    answer: "No. If your flows run on a bench or power a low-stakes internal tool, the open community library serves you well and managing your own vetting is a reasonable choice. Certified Nodes matter most when your flows touch sensitive data, fall under regulation, carry uptime obligations, or back a team that can't afford to debug alone."
  - question: "How do Certified Nodes help in regulated environments?"
    answer: "Regulators expect evidence that each node met a standard, not an assurance that it 'seemed fine.' Because every Certified Node has cleared defined quality and security checks and its certification status is tracked and communicated, certification produces the audit trail by default rather than leaving you to reconstruct provenance after the fact."
  - question: "How do Certified Nodes reduce downtime?"
    answer: "Much unplanned downtime comes from a node that ran fine for months breaking when an update shifts an underlying behavior. FlowFuse tests Certified Nodes for compatibility against the version they're meant to run on, which removes that class of surprise-upgrade failure — valuable for any operation measured in nines of availability."

>   - question: "What happens if a node loses certification?"

Reviewer 3: @KristopherLeads is this even possible? It thought it wasn't...

Author: Which part specifically?

Reviewer 3: > a node loses certification?

Author: Unclear, but I excised it to be safe.

Reviewer 3: ok, but it would be good to know, even internally... Have you asked the team?

  - question: "Which FlowFuse tiers include Certified Nodes?"
    answer: "FlowFuse Certified Nodes are available to Teams and Enterprise tier customers. New instances get automatic access to the catalogue, and you can contact FlowFuse to add Certified Nodes to an existing instance."
tldr: "Every node in a flow is code running in your environment, and community nodes carry hidden security, compatibility, and maintenance risk you have to manage yourself. FlowFuse Certified Nodes take that work on through a defined quality, security, and support process. They're the clear solve for four situations — high-security deployments, regulated environments, uptime-critical operations, and teams that can't afford to debug alone — and they're available to Teams and Enterprise tier customers."
cta:
  type: contact
  title: "Not sure if Certified Nodes fit your deployment?"
  description: "See how FlowFuse Certified Nodes take the vetting, security monitoring, and support burden off your team. Talk to us about your setup and which use case describes you."
---

# Why FlowFuse Certified Nodes?

> With every flow that you build on FlowFuse, you’re using nodes to execute code in your environment. These nodes often connect to your data, your network, and your operations - and while the flexibility and freedom these nodes offer are part of the power behind FlowFuse, the reality of adoption means that you’re inheriting functionality as much as you are inheriting risk. Open source development is often offered by people who assume no liability. They needed a solution, they developed the solution, and it worked, but when it comes to adoption, the phrase "caveat emptor" - buyer beware - applies.

Reviewer 2: ```suggestion
Every flow you build on FlowFuse runs on nodes, and each node is code executing in your environment, often with access to your data, your network, and your operations. That flexibility is part of what makes FlowFuse powerful. It also means you take on some risk with each node you install.
```

> For most users, this risk is minimal - but for some orgs, the idea of inheriting risk is a step too far. When an organisation cannot have a day of downtime following a code update, or a potential risk vector unaddressed after a maintainer loses interest, something more is necessary - and FlowFuse Certified Nodes presents a solution to this problem.

Reviewer 2: ```suggestion
Most open source nodes are built by individual developers who assume no liability. They solved a problem for themselves, built the solution, and it worked, but they took on no obligation to maintain it for you. As a user, the vetting responsibility falls on you. For most users, that risk is minor and manageable. But for some organizations, it isn't: when a day of downtime isn't an option, or a security issue can't sit unaddressed because a maintainer has moved on, you need more than "it worked when we checked."
```

> FlowFuse Certified nodes exist to take that liability off your hands, delivering functionality and peace of mind. A Certified Node is one that FlowFuse has put through an intense quality and security process, ensuring that it meets form and function before you ever install it. And once it is installed, FlowFuse offers long-term support and critical review capability - meaning that the vetting, monitoring, and maintenance you’d otherwise need to carry becomes something that FlowFuse can solve for you.

Reviewer 2: ```suggestion
That's what FlowFuse Certified Nodes are for. A Certified Node is one FlowFuse has put through a quality and security process before it reaches your palette. Once installed, FlowFuse offers long-term support and ongoing review, so the vetting, monitoring, and maintenance you'd otherwise carry yourself becomes work FlowFuse does for you.
```

<!--more-->

## What "Certified" Actually Means

FlowFuse certifies a node against three pillars, each aimed at a specific failure mode.

### Quality

> Every Certified Node is tested for operational reliability and compatibility, so it behaves predictably across versions instead of surprising you in production. This means no more guesswork, no more deep vetting, and significantly quicker time to value. We pick only the best-in-class OSS nodes, sub-license best-in-class properietary nodes, and build custom integrations for FlowFuse users. This means you get true best-in-class integrations in a single catalog - with no need to select new vendors or navigate complex licenses.

Reviewer 2: ```suggestion
Every Certified Node is tested for reliability and compatibility, so it behaves consistently across versions and in production. This reduces the vetting work on your end and shortens time to value. We select proven open-source nodes, sub-license proprietary nodes where it makes sense, and build custom integrations when needed. The result is one catalogue of dependable integrations, without having to evaluate new vendors or manage separate licenses.
```

Reviewer 3: I'd recommend replacing "time to value" with something like "Get to production faster" or "Build with confidence from day one"

"Time to value" feels like a very SaaS-centric term. I think a more concrete outcome would resonate better with our audience.

### Security
FlowFuse implements secure design and architecture by default, and this process extends to the Certified Nodes platform. Every node is checked, validated, and then secured via ongoing reviews per an SLA. This means that you don't stress, especially when fixes are time sensitive, converting a worry-mode update-fest to a smooth 5-minute task.

### Support

> When a Certified Node gives you trouble, you get real troubleshooting help and a defined path to resolution, not an unanswered issue queue. FlowFuse node certification also means FlowFuse supported - so you get real, human support, not automatically generated documentation that may or may not map to solutions.

Reviewer 2: ```suggestion
When a Certified Node causes an issue, FlowFuse provides troubleshooting help and a clear path to resolution, rather than an open-ended issue queue. Certification also means FlowFuse support, so you're working with a person who can help, not just documentation that may or may not apply to your situation.
```

## How to Know If This Is You

These four situations aren't mutually exclusive - and they’re by no means exhaustive. A hospital system can carry all four at once - security, compliance, uptime, and support - as well as a complex maze of additional operational requirements and restrictions. FlowFuse Certified Nodes are meant to resolve this complexity and deliver a powerful implementation at scale, resulting in a stack that is deployable, maintainable, and - critically - trustworthy.

If your flows run on a bench or power a low-stakes internal tool, the open library serves you well, and managing your own vetting is a perfectly reasonable choice. Not every team needs Certified Nodes, and we won't pretend otherwise. 

But if your flows touch sensitive data, fall under industry regulations, carry any uptime obligations, or back a team that can't afford to debug alone, then the vetting, monitoring, and support you'd otherwise have to build yourself is exactly what FlowFuse provides. The value is not just in the nodes themselves - it's in handing off work that was quietly becoming yours by default.

## Certified Ndoes Today

> Right now, FlowFuse offers [[Certified Nodes](/integrations/?certified=1)](/integrations/?certified=1) across a focused, high-value set of integrations, split across two categories: FlowFuse Edge and FlowFuse Hub. This includes official support for [[Redis](/node-red/flowfuse/hub/redis)](/node-red/flowfuse/hub/redis) under Hub, and the [[CIP Suite](/node-red/flowfuse/edge/cip-suite)](/node-red/flowfuse/edge/cip-suite) for Allen-Bradley/EtherNet-IP devices, [[OPC UA](/node-red/flowfuse/edge/opcua/)](/node-red/flowfuse/edge/opcua/), and [[RTSP](/node-red/flowfuse/edge/rtsp/)](/node-red/flowfuse/edge/rtsp/) under Edge. We're actively expanding this catalogue, with major additions planned in the coming weeks to cover even more devices and systems.

Reviewer 3: @KristopherLeads the links format is wrong in this section. Check out how it's rendering:
<img width="720" height="206" alt="Image" src="https://github.com/user-attachments/assets/da9b9d18-bbdd-4f7e-b4c3-ee76bcfb82ad" />

Reviewer 2: Fixed

Reviewer 2: https://github.com/FlowFuse/website/pull/5152/changes/d9b8b54f1654efe92e44321352c6171369f4aca2

> Right now, FlowFuse offers [[Certified Nodes](/integrations/?certified=1)](/integrations/?certified=1) across a focused, high-value set of integrations, split across two categories: FlowFuse Edge and FlowFuse Hub. This includes official support for [[Redis](/node-red/flowfuse/hub/redis)](/node-red/flowfuse/hub/redis) under Hub, and the [[CIP Suite](/node-red/flowfuse/edge/cip-suite)](/node-red/flowfuse/edge/cip-suite) for Allen-Bradley/EtherNet-IP devices, [[OPC UA](/node-red/flowfuse/edge/opcua/)](/node-red/flowfuse/edge/opcua/), and [[RTSP](/node-red/flowfuse/edge/rtsp/)](/node-red/flowfuse/edge/rtsp/) under Edge. We're actively expanding this catalogue, with major additions planned in the coming weeks to cover even more devices and systems.

Reviewer 2: ```suggestion
Right now, FlowFuse offers [Certified Nodes](/integrations/?certified=1) across a focused, high-value set of integrations, split across two categories: FlowFuse Edge and FlowFuse Hub. This includes official support for [Redis](/node-red/flowfuse/hub/redis) under Hub, and the [CIP Suite](/node-red/flowfuse/edge/cip-suite) for Allen-Bradley/EtherNet-IP devices, [OPC UA](/node-red/flowfuse/edge/opcua/) and [RTSP](/node-red/flowfuse/edge/rtsp/) under Edge. We're actively expanding this catalogue, with major additions planned in the coming weeks to cover even more devices and systems.
```

> Right now, FlowFuse offers [Certified Nodes](https://flowfuse.com/integrations/?certified=1) across a wide variety of functions, offering official support for solutions like [ModBus](https://flowfuse.com/integrations/node-red-contrib-modbus/), [MongoDB](https://flowfuse.com/integrations/node-red-contrib-mongodb4/), [MCP](https://flowfuse.com/integrations/node-red-contrib-mcprotocol/), [OPC UA](https://flowfuse.com/node-red/flowfuse/edge/opcua/), and much more. We are constantly adding to this catalogue - with major additions planned in the coming weeks to expand this offering to even more devices and systems.

Reviewer 2: These are the recommended nodes. Certified Nodes v2 includes a different set of nodes. You can check them here: https://flowfuse.com/integrations/

> Right now, FlowFuse offers [Certified Nodes](https://flowfuse.com/integrations/?certified=1) across a wide variety of functions, offering official support for solutions like [ModBus](https://flowfuse.com/integrations/node-red-contrib-modbus/), [MongoDB](https://flowfuse.com/integrations/node-red-contrib-mongodb4/), [MCP](https://flowfuse.com/integrations/node-red-contrib-mcprotocol/), [OPC UA](https://flowfuse.com/node-red/flowfuse/edge/opcua/), and much more. We are constantly adding to this catalogue - with major additions planned in the coming weeks to expand this offering to even more devices and systems.

Reviewer 2: ```suggestion
Right now, FlowFuse offers [[Certified Nodes](/integrations/?certified=1)](/integrations/?certified=1) across a focused, high-value set of integrations, split across two categories: FlowFuse Edge and FlowFuse Hub. This includes official support for [[Redis](/node-red/flowfuse/hub/redis)](/node-red/flowfuse/hub/redis) under Hub, and the [[CIP Suite](/node-red/flowfuse/edge/cip-suite)](/node-red/flowfuse/edge/cip-suite) for Allen-Bradley/EtherNet-IP devices, [[OPC UA](/node-red/flowfuse/edge/opcua/)](/node-red/flowfuse/edge/opcua/), and [[RTSP](/node-red/flowfuse/edge/rtsp/)](/node-red/flowfuse/edge/rtsp/) under Edge. We're actively expanding this catalogue, with major additions planned in the coming weeks to cover even more devices and systems.
```

> FlowFuse Certified Nodes are available to Teams and Enterprise tier customers. New instances get automatic access to the catalogue, and you can [contact us](https://flowfuse.com/contact-us) to add them to an existing instance - or to talk through which of these problems you're facing. Reach out to us today to see if FlowFuse Certified Nodes can help you and your stack!

Reviewer 2: In blogs or website pages, you don't need to use the full URL. Just use the relative path

```suggestion
FlowFuse Certified Nodes are available to Teams and Enterprise tier customers. New instances get automatic access to the catalogue, and you can [contact us](/contact-us) to add them to an existing instance - or to talk through which of these problems you're facing. Reach out to us today to see if FlowFuse Certified Nodes can help you and your stack!
```

Reviewer 2: cc @KristopherLeads

> FlowFuse Certified Nodes are available to Teams and Enterprise tier customers. New instances get automatic access to the catalogue, and you can [contact us](https://flowfuse.com/contact-us) to add them to an existing instance - or to talk through which of these problems you're facing. Reach out to us today to see if FlowFuse Certified Nodes can help you and your stack!

Reviewer 2: This needs to be updated to reflect the new offering
